python/cudf/cudf/utils/queryutils.py

```diff
@@ -220,7 +220,7 @@ def query_execute(df, expr, callenv):
             "or bool dtypes."
         )
 
-    colarrays = [col.data_array_view for col in colarrays]
+    colarrays = [col.data_array_view(mode="write") for col in colarrays]
```

Comment:
These columns are only read in `make_aggregate_nullmask`, no? If yes, do we also need to acquire a spill lock?

Reply:
Yup, this can be a read-only with spill lock. Made the change 👍